File: module_beauty/src/main/java/com/gas/beauty/ui/main/mvp/MainModel.java
package com.gas.beauty.ui.main.mvp;

import com.base.lib.di.scope.ActivityScope;
import com.base.lib.integration.repository.IRepositoryManager;
import com.base.lib.mvp.BaseModel;
import com.gas.beauty.bean.GankBaseResponse;
import com.gas.beauty.bean.GankItemBean;
import com.gas.beauty.http.GankService;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Observable;

@ActivityScope
public class MainModel extends BaseModel implements MainContract.Model {


    @Inject
    public MainModel(IRepositoryManager repositoryManager) {
        super(repositoryManager);
    }


    @Override
    public Observable<GankBaseResponse<List<GankItemBean>>> getGirlList(int num, int page) {
        return mRepositoryManager.obtainRetrofitService(GankService.class)
                .getGirlList(num,page);
    }
}

File: module_beauty/src/main/java/com/gas/beauty/ui/main/mvp/MainModel.kt
package com.gas.beauty.ui.main.mvp

import com.base.lib.di.scope.ActivityScope
import com.base.lib.integration.repository.IRepositoryManager
import com.base.lib.mvp.BaseModel
import com.gas.beauty.bean.GankResponse
import com.gas.beauty.bean.BeautyBean
import com.gas.beauty.http.GankService
import io.reactivex.Observable
import javax.inject.Inject

@ActivityScope
class MainModel @Inject constructor(repositoryManager: IRepositoryManager) : BaseModel(repositoryManager), MainContract.Model {
    override fun getGirlList(num: Int, page: Int): Observable<GankResponse<List<BeautyBean>>> {
        return mRepositoryManager.obtainRetrofitService(GankService::class.java)
                .getGirlList(num, page)
    }
}